sdp/processors/huggingface/speech_recognition.py

```diff
@@ -0,0 +1,127 @@
+        batch_size (str): Inference batch size.
+    """
+    def __init__(
+        self,
+        pretrained_model: str,
+        output_text_field: str,
+        device: str = None,
+        batch_size: str = 1,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        import torch
+        import whisper # pip install -U openai-whisper
+        self.whisper = whisper
+        self.pretrained_model = pretrained_model
+        self.output_text_field = output_text_field
+        self.device = device
+        self.batch_size = batch_size
+        if self.device is None:
+            if torch.cuda.is_available():
+                self.device = "cuda"
+            else:
+                self.device = "cpu"
+        self.model = whisper.load_model(self.pretrained_model)
+    
+    def process(self):
+        json_list = load_manifest(Path(self.input_manifest_file))
+        
+        Path(self.output_manifest_file).parent.mkdir(exist_ok=True, parents=True)
+        
+        with Path(self.output_manifest_file).open('w') as f:
+            for item in tqdm(json_list):
+                pred_text, pred_lang = self.whisper_infer(item["audio_filepath"])
+
+                item[self.output_text_field] = pred_text
+                f.write(json.dumps(item, ensure_ascii=False) + '\n')
+
+    def whisper_infer(self, audio_path):
+        audio = self.whisper.load_audio(audio_path)
+
+        audio = self.whisper.pad_or_trim(audio)
+        mel = self.whisper.log_mel_spectrogram(audio)
+        mel = mel.to(self.device)
+
+        _, probs = self.model.detect_language(mel)
+        lang = max(probs, key=probs.get)
+        
+        options = self.whisper.DecodingOptions()
+        result = self.whisper.decode(self.model, mel, options)
+        return result.text, lang
+    
+class ASRTransformer(BaseProcessor):
+    """
+    Processor to transcribe using ASR Transformer model from HuggingFace.
+    
+    Args:
+        pretrained_model (str): name of pretrained model on HuggingFace.
+        output_text_field (str): field to save transcription result.
+        device (str): Inference device.
+        batch_size (str): Inference batch size.
+    """
+    def __init__(
+        self,
+        pretrained_model: str,
+        output_text_field: str,
+        device: str = None,
+        batch_size: str = 1,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        import torch
+        from transformers import AutoModelForSpeechSeq2Seq, AutoProcessor, pipeline
+        
+        self.pretrained_model = pretrained_model
+        self.output_text_field = output_text_field
+        self.device = device
+        self.batch_size = batch_size
+        if self.device is None:
+            if torch.cuda.is_available():
+                self.device = "cuda:0"
+            else:
+                self.device = "cpu"
+        
+        torch_dtype = torch.float16 if torch.cuda.is_available() else torch.float32
```

Comment:
probably best to access dtype as a parameter, since float16 might not always be the best option

Reply:
added